feat: Document how to use SQLite3 and remove PostgreSQL-only mentions (#524)
serverpod/serverpod_docs: docs/06-concepts/02-models/03-vector-fields.md

To ensure optimal performance with vector similarity searches, consider creating specialized vector indexes on your vector fields. See the [Vector indexes](../database/indexing#vector-indexes) section for more details.

:::info

> The usage of Vector fields requires the PostgreSQL database with the `pgvector` extension installed - which comes by default on new Serverpod projects. To upgrade an existing project, see the [Upgrading to pgvector support](../../upgrading/upgrade-to-pgvector) guide.

Reviewer: Nit: ` - ` is reading as an em-dash stand-in, and "which" is ambiguous (extension, database, or projects?).

```suggestion
The usage of Vector fields requires a PostgreSQL database with the `pgvector` extension installed. The extension comes by default on new Serverpod projects. To upgrade an existing project, see the [Upgrade to pgvector](../../upgrading/archive/upgrade-to-pgvector) guide.
```

Author: Fixed at https://github.com/serverpod/serverpod_docs/pull/524/commits/fbc4af43e141624ebd49157c2f3b4cbe02c4818e.
